### PR TITLE
impl(generator): add (unimplemented) tracing connection

### PR DESCRIPTION
**Pull request by dbolduc**
Part of the work for #10491

Add a tracing connection that just passes through to its child connection. I don't return `kUnimplemented`, because I want a partially implemented tracing connection decorator to still be functional.

I will later introduce a `MakeFooTracingConnnection(...)` factory function, which will be declared outside of the `#ifdef` guards.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10526)

### CODE REVIEW

File: generator/generator_test.cc [hunk 1]
```diff
@@ -141,7 +141,7 @@ TEST_F(GeneratorTest, BadCommandLineArgs) {
 }
 
 TEST_F(GeneratorTest, GenerateServicesSuccess) {
-  int constexpr kNumMockOutputStreams = 23;
+  int constexpr kNumMockOutputStreams = 25;
```
Comment (devbww) [inline generator/generator_test.cc:144]: That looks unfortunate.

File: generator/internal/tracing_connection_generator.h [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  Status GenerateCc() override;
+
+  static std::string MethodDeclaration(
+      google::protobuf::MethodDescriptor const& method);
+  static std::string AsyncMethodDeclaration(
+      google::protobuf::MethodDescriptor const& method);
+  static std::string MethodDefinition(
+      google::protobuf::MethodDescriptor const& method);
+  static std::string AsyncMethodDefinition(
+      google::protobuf::MethodDescriptor const& method);
```
Comment (devbww) [inline generator/internal/tracing_connection_generator.h:56]: Do these need to be members?

File: generator/integration_tests/golden/v1/internal/golden_kitchen_sink_tracing_connection.cc [hunk 3]
```diff
@@ -0,0 +1,90 @@
+// Copyright 2022 Google LLC
```
Comment (devbww) [inline generator/integration_tests/golden/v1/internal/golden_kitchen_sink_tracing_connection.cc:1]: Where did these years originate?

File: generator/internal/dern.cc [hunk 4]
```diff
@@ -0,0 +1,5 @@
+/// Returns `child` if tracing is not enabled.
```
Comment (devbww) [inline generator/internal/dern.cc:1]: So what?

File: generator/generator_test.cc (hunk 1 above)
Comment (dbolduc) [reply generator/generator_test.cc:144]: Eh, it's not so bad. I'd rather have the test than not.

File: generator/internal/tracing_connection_generator.h (hunk 2 above)
Comment (dbolduc) [reply generator/internal/tracing_connection_generator.h:56]: No. But it is consistent with what we do for the `ConnectionImpl` https://github.com/googleapis/google-cloud-cpp/blob/9196bdcd6281afb62456b648ca436cdfddaf6241/generator/internal/connection_impl_generator.h#L49-L56

(It's probably the laziness talking, but) I'd rather change neither than both.

File: generator/integration_tests/golden/v1/internal/golden_kitchen_sink_tracing_connection.cc (hunk 3 above)
Comment (dbolduc) [reply generator/integration_tests/golden/v1/internal/golden_kitchen_sink_tracing_connection.cc:1]: https://github.com/googleapis/google-cloud-cpp/blob/9196bdcd6281afb62456b648ca436cdfddaf6241/generator/integration_tests/golden_config.textproto#L21

File: generator/internal/dern.cc (hunk 4 above)
Comment (dbolduc) [reply generator/internal/dern.cc]: Gah, thanks for catching this. `dern.*` is a pejorative version of my name that I use for temporary files. I did not mean to commit it. Removed.

File: generator/integration_tests/golden/v1/internal/golden_kitchen_sink_tracing_connection.cc (hunk 3 above)
Comment (devbww) [reply generator/integration_tests/golden/v1/internal/golden_kitchen_sink_tracing_connection.cc:1]: Hmm ... we add new output files to the generator and they appear from the past.  Spooky.

**Review by devbww — APPROVED** [review]

**Pull request merged**